sktime/utils/deep_equals/__init__.py

```diff
@@ -0,0 +1,19 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Module for nested equality checking."""
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+# todo 0.26.0: check whether scikit-base>=0.6.1 lower bound is 0.6.1 or higher
+# if yes, remove legacy handling and only use the new deep_equals
+if _check_soft_dependencies(
+    "scikit-base<0.6.1",
+    package_import_alias={"scikit-base": "skbase"},
+    severity="none",
+):
+    from sktime.utils._testing.deep_equals import deep_equals
```

Comment:
Probably a naive question, but why does it need a condition on skbase version to import sktime internal function?

Reply:
please let me know if the following explanation is not clear enough, the "why" is a bit tricky and has to do with deprecation.

Primary reason is that the `skbase` `deep_equals` has the option to accommodate plugins (such as for `ForecastingHorizon` or `polars`) only since `skbase` version 0.6.1.

So, if a user would have an earlier version of `skbase` installed, without the case distinction `sktime` testing and some `sktime` user features would error out for them.

So, one way to make earlier versions of `skbase` compatible is to make a case distinction and ensure there is a compatible `deep_equals` if `skbase` version is 0.6.0 or lower.

Why imo we shouldn't just raise the lower bound for `skbase`: corporate ecosystems, especially with higher security requirements, have approval processes that might lag behind months or years for packages and versions. `sktime` is likely to get approved earlier than `skbase`, due to footprunt.